src/main/java/com/mathworks/ci/TestResultsViewAction.java

```diff
@@ -0,0 +1,265 @@
+                Object jsonTestSessionResults = testArtifactIterator.next();
+
+                List<MatlabTestFile> testSessionResults = new ArrayList<>();
+                Map<String, MatlabTestFile> map = new HashMap<>();
+
+                if(jsonTestSessionResults instanceof JSONArray){
+                    JSONArray jsonTestSessionResultsArray = (JSONArray) jsonTestSessionResults;
+                    Iterator<JSONObject> testSessionResultsIterator = jsonTestSessionResultsArray.iterator();
+
+                    while(testSessionResultsIterator.hasNext()){
+                        JSONObject jsonTestCase = testSessionResultsIterator.next();
+                        getTestSessionResults(testSessionResults, jsonTestCase, map);
+                    }
+                }
+                else if(jsonTestSessionResults instanceof JSONObject) {
+                    JSONObject jsonTestCase = (JSONObject) jsonTestSessionResults;
+                    getTestSessionResults(testSessionResults, jsonTestCase, map);
+                }
+
+                testResults.add(testSessionResults);
+            }
+        }
+        catch (Exception e) {
+            throw new IOException(e.getLocalizedMessage());
+        }
+
+        return testResults;
+    }
+
+    private void getTestSessionResults(List<MatlabTestFile> testSessionResults, JSONObject jsonTestCase, Map<String, MatlabTestFile> map) throws IOException, InterruptedException {
+        FilePath baseFolder = new FilePath(new File(jsonTestCase.get("BaseFolder").toString()));
+        JSONObject matlabTestCaseResult = (JSONObject) jsonTestCase.get("TestResult");
+
+        // Not OS dependent
+        String[] testNameSplit = matlabTestCaseResult.get("Name").toString().split("/");
+        String matlabTestFileName = testNameSplit[0];
+        String matlabTestCaseName = testNameSplit[1];
+
+        // Check if test's file was known or not
+        MatlabTestFile matlabTestFile = map.get(baseFolder + File.separator + matlabTestFileName);
+        if(matlabTestFile == null) {
+            matlabTestFile = new MatlabTestFile();
+            matlabTestFile.setName(matlabTestFileName);
+
+            map.put(baseFolder + File.separator + matlabTestFileName, matlabTestFile);
+            testSessionResults.add(matlabTestFile);
+        }
+
+        // Calculate the relative path
+        Path path1 = Paths.get(baseFolder.toURI());
+        Path path2 = Paths.get(this.workspace.toURI());
+        Path relPath = path2.relativize(path1);
+
+        matlabTestFile.setPath(this.workspace.getName() + File.separator + relPath.toString());
+
+        MatlabTestCase matlabTestCase = new MatlabTestCase();
+        matlabTestCase.setName(matlabTestCaseName);
+        matlabTestCase.setPassed((boolean) matlabTestCaseResult.get("Passed"));
+        matlabTestCase.setFailed((boolean) matlabTestCaseResult.get("Failed"));
+        matlabTestCase.setIncomplete((boolean) matlabTestCaseResult.get("Incomplete"));

```

Comment:
Maybe the `MatlabTestCase` constructor can take the name as well (as Sam suggested above for `MatlabTestFile`). I am wondering if we can just derive and set the status directly here and get rid of those boolean properties on `MatlabTestCase` or if we're also using them somewhere else.

Reply:
Something like
```
if (matlabTestCaseResult.get("Passed")) {
    matlabTestCase.setStatus(MatlabBuilderConstants.PASSED)
} else if (matlabTestCaseResult.get("Failed")) {
    matlabTestCase.setStatus(MatlabBuilderConstants.FAILED)
} else if (matlabTestCaseResult.get("Incomplete")) {
    matlabTestCase.setStatus(MatlabBuilderConstants.INCOMPLETE)
}
```
It's a little messier here but it's pretty much the same as what's in the `updateStatus` method.

Reply:
Updated the constructor, thanks! Rest of the part is addressed in the above comment